hearth/lib/hearth/middleware/send.rb

```diff
@@ -79,37 +69,99 @@ def call(input, context)
 
       private
 
+      def stub_response(input, context, output)
+        stub = @stubs.next(context.operation_name)
+        apply_stub(stub, input, context, output)
+        return unless context.response.body.respond_to?(:rewind)
+
+        context.response.body.rewind
+      end
+
+      def send_request(context, output)
+        @client.transmit(
+          request: context.request,
+          response: context.response,
+          logger: context.logger
+        )
+      rescue Hearth::NetworkingError => e
+        output.error = e
+      end
+
       def apply_stub(stub, input, context, output)
         case stub
         when Proc
           stub = stub.call(input, context)
           apply_stub(stub, input, context, output) if stub
-        when Exception
+        when Exception, ApiError
           output.error = stub
-        when Class
-          output.error = stub.new
         when Hash
-          @stub_class.stub(
-            context.response,
-            stub: @params_class.build(
-              stub,
-              context: 'stub'
-            )
-          )
+          apply_stub_hash(stub, context)
         when NilClass
-          @stub_class.stub(
-            context.response,
-            stub: @params_class.build(
-              @stub_class.default,
-              context: 'stub'
-            )
-          )
+          apply_stub_nil(context)
+        when Hearth::Structure
+          apply_stub_hearth_structure(stub, context)
         when Hearth::Response
           context.response.replace(stub)
         else
           raise ArgumentError, 'Unsupported stub type'
         end
       end
+
+      def apply_stub_hash(stub, context)
+        if stub.key?(:error)
+          apply_stub_hash_error(stub, context)
+        elsif stub.key?(:data)
+          apply_stub_hash_data(stub, context)
+        else
```

Comment:
Should we validate in the above cases that it contains ONLY error or data? Right now you could pass {error: ..., data: ...}.  Or something like {error: MyError, message: 'I think this will work but it gets ignored'}

Reply:
Additionally - do we want validation when adding the stubs on the client? That puts the error close to when its created.  But obviously we'd still need the same validation here since you can stub with a proc and that could return an invalid stub.  So to avoid repeating.... maybe we only do validation at this point?

Reply:
Validated it's either one or the other. I think we don't want to validate when adding stubs because of procs - late as possible is fine I think.